app/actions/websocket/teams.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DeviceEventEmitter} from 'react-native';
+
+import {localRemoveUserFromTeam} from '@actions/local/team';
+import {updateUsersNoLongerVisible} from '@actions/remote/user';
+import {queryCurrentTeamId} from '@app/queries/servers/system';
+import {queryCurrentUser} from '@app/queries/servers/user';
+import {isGuest} from '@app/utils/user';
+import DatabaseManager from '@database/manager';
+
+export async function handleLeaveTeamEvent(serverUrl: string, msg: any) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const currentTeamId = await queryCurrentTeamId(database.database);
+    const user = await queryCurrentUser(database.database);
+    if (!user) {
+        return;
+    }
+
+    if (user.id === msg.data.user_id) {
+        localRemoveUserFromTeam(serverUrl, msg.data.team_id);
+
+        // TODO originally it also removed channel categories and threads in teams. Apparently none of those features are yet implemented.
+        // TODO consider whether localRemove should also remove team channel history and team search history.
```

Comment:
this in fact seems to be removed when calling `localRemoveUserFromTeam` which is an action function that I don't really think we need and should be handled by this action instead.